src/event_handler.rs

```diff
@@ -508,151 +457,192 @@ pub async fn handle_event_logs(
-                    let error_string = format!("{:?}", pending_txn.unwrap_err());
+            let pending_txn = pending_txn_data
+                .http_rpc_client
+                .send_transaction(dummy_replacement_txn, None)
+                .await;
+            let Ok(pending_txn) = pending_txn else {
+                let error_string = format!("{:?}", pending_txn.unwrap_err());
+                eprintln!(
+                    "Failed to send the dummy replacement txn for the nonce {}: {}",
+                    pending_txn_data.nonce, error_string
+                );
 
-                    // Retry after a delay if connection failed
-                    if error_string.contains("code: -32000")
-                        && (error_string.contains("connection") || error_string.contains("network"))
-                    {
-                        sleep(Duration::from_millis(200)).await;
+                // Handle retry logic for the dummy txn
+                match parse_send_error(error_string.to_lowercase()) {
+                    JobsTxnSendError::NonceTooLow => {
+                        // Txn mined for the current nonce and hence no need to retry
+                        break;
+                    }
+                    JobsTxnSendError::OutOfGas => {
+                        pending_txn_data.gas_limit = pending_txn_data.gas_limit + GAS_LIMIT_BUFFER;
                         continue;
                     }
-
-                    // Retry with a higher gas price if the replacement txn is underpriced
-                    if error_string.contains("code: -32000")
-                        && (error_string.contains("replacement transaction underpriced")
-                            || error_string.contains("transaction underpriced"))
-                    {
+                    JobsTxnSendError::GasPriceLow => {
                         pending_txn_data.gas_price =
                             U256::from(100 + RESEND_GAS_PRICE_INCREMENT_PERCENT)
                                 * pending_txn_data.gas_price
                                 / 100;
-
-                        sleep(Duration::from_millis(10)).await;
                         continue;
                     }
+                    _ => {
+                        sleep(Duration::from_millis(200)).await;
+                        continue;
+                    }
+                }
+            };
 
-                    // Break in case of other errors like "nonce too low" or "execution reverted" or "insufficient funds" etc.
-                    retry_txn = false;
-                    break;
-                };
+            // Wait for confirmation of the sent txn
+            let Ok(Some(_)) = pending_txn
+                .confirmations(1)
+                .interval(Duration::from_secs(1))
+                .await
+            else {
+                // If even this simple dummy txn fails to be mined then check whether the operator has injected a new gas account to send the txns
```

Comment:
There is another way to do all this.

When gas address is getting changed, hold the write lock on pending_txn_deque, update it's all elements with new httpclient and nonce. Then just let other threads running as it is.

Reply:
new nonce corresponding to new gas address.

My idea, you still wait for old txn_hash to be included, but if you need to send replacement transaction then send it with latest gas address and corresponding nonce.

To implement this behavior, we can do the necessary updates in pending_txn to use new nonce and gas address, when mutable config is updated.

Reply:
Sending the replacement txn with new nonce and gas account will create problems at L516. I think it's easier to move the pending_txns with new nonce to the main txn sender `send_execution_output` where the nonce handling will be also handled and if the pending txn went through with previous account then the replacement txn will just exit at L238 due to `JobsExecutorAlreadySubmittedOutput` error during execution.

Reply:
are you referring to this line ` JobsTxnSendError::NonceTooLow` ? What exactly the problem would be?

You understand the behavior I mentioned above. Do you agree with that?

Reply:
I agree with the behavior that when the a new gas account is injected we want to send the replacement txns with that account itself. The problem with sending those txns in the `resend_pending_transaction` thread is that we're keeping track of the `nonce_to_send` for the next transaction in order to maintain consistency and we're incrementing it only when a transaction is sent successfully to the network. Let's say we have M pending txns in the queue and a new gas account is injected then according to your approach we'll get the current nonce `N` for that address and update the nonces of the pending txns like `N`, `N+1`, `N+2`, ... and therefore the `nonce_to_send` will have to be updated to `N+M` for sending txns simultaneously without waiting for previous nonces to be mined. But what if pending txns upto `1=<X<=M` in the queue went through with the old account before we could resend them with the new one and hence our thread won't resend those txns keeping the nonces `N`....`N+X-1` for the new account idle. This will create a nonce gap for the new gas account txns and the remaining txns that we send with it will go through and stay in the mempool forever. To avoid these kind of nonce issues and clashes I'm suggesting that we send all the new nonce txns in only one thread i.e., the `send_execution_output`. Apart from monitoring the pending txns our main focus is also on unblocking the previous pending nonces so that the new ones can be mined.